### PR TITLE
It2 game avg goals by season

### LINKED ISSUE #18: Game stats - average goals by season
Description:
Average number of goals scored in a game organized in a hash with season names (e.g. 20122013) as keys and a float representing the average number of goals in a game for that season as values (rounded to the nearest 100th)

Result:
Hash

### PR DESCRIPTION
**Pull request by leahriffell**
**Helper methods added:** 
- seasons: returns array of unique seasons within game fixture file 
- array_of_total_goals_per_season: returns hash with seasons as keys and array of goals for each game as values 
- total_goals_per_season: uses above method to return hash of seasons as keys and sum of the array of goals above as values (maybe these could potentially be consolidated?)

**Stat method added:** 
- average_goals_by_season

closes #18

### CODE REVIEW

**Review by sarynm12 — APPROVED** [review]
Looks great, Leah! Thank you!

File: lib/stat_tracker.rb [hunk 1]
```diff
@@ -23,16 +23,33 @@ def initialize(game_path, team_path, game_teams_path)
...
+    end
+  end
+
+  def total_goals_per_season
+    result = {}
+    array_of_total_goals_per_season.each do |season, goals|
+      result[season] = goals.sum
+    end
+    result
+  end
```
Comment (AngelaGuardia) [inline lib/stat_tracker.rb]: I think these two methods could just be one like this:

```
def total_goals_per_season
    result = Hash.new(0)
    @games.each do |game|
      result[game.season] += game.away_goals + game.home_goals
    end
    result
  end
```

Unless you will use the breakdown given by `array_of_total_goals_per_season`. The above could also be refactored using reduce.

**Review by AngelaGuardia — CHANGES_REQUESTED** [review]
This looks really good Leah. Just one suggestion!

File: lib/stat_tracker.rb (hunk 1 above)
Comment (leahriffell) [reply lib/stat_tracker.rb]: @guardiac yesssss, thank you! I've refactored this down to one method using .reduce.  I don't have a need for the breakdown from `array_of_total_goals_per_season`.

File: lib/stat_tracker.rb [hunk 2]
```diff
@@ -35,19 +35,12 @@ def total_goals_per_game
...
-  end
-
   def total_goals_per_season
-    result = {}
-    array_of_total_goals_per_season.each do |season, goals|
-      result[season] = goals.sum
+    @games.reduce({}) do |result, game|
+      result[game.season] = 0 if !result[game.season]
+      result[game.season] += game.away_goals + game.home_goals
+      result
```
Comment (AngelaGuardia) [inline lib/stat_tracker.rb]: Nice Leah! Another way to do this and avoid the guard statement is with `.reduce(Hash.new(0))`. I think is perfect tho! No need to change.

**Review by AngelaGuardia — APPROVED** [review]

File: lib/stat_tracker.rb (hunk 2 above)
Comment (leahriffell) [reply lib/stat_tracker.rb]: Ooh, good call! I think having a default value of 0 here is more readable so I ended up updating it. Thanks for the suggestion!

**Review by hopesgit — APPROVED** [review]

**Pull request merged**